### PR TITLE
chore(test cases): add playwright for e2e tests

### LINKED ISSUE #131: integration of Playwright for e2e tests
integration of Playwright for e2e tests

### PR DESCRIPTION
**Pull request by DeepakSF25**
## Description
Implemented Playwright E2E testing with project-specific configurations in e2e folder for arc-docs, arc, and saas-ui projects, each with isolated test suites, dedicated ports, and independent execution capabilities.

Fixes # ([131](https://github.com/sourcefuse/angular-boilerplate/issues/131))

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested locally

## Checklist:

- [x] Performed a self-review of my own code
- [x] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by rohit-sourcefuse** [comment]
This PR lays out a strong foundation for Playwright E2E across multiple Angular apps with clean configuration and project separation. However, a few things should be addressed to make the E2E setup robust and future-proof:

**1. Missing Tests for arc and saas-ui**

There's no working test suite or even a basic smoke test for `arc` and `saas-ui`. Please add something like:

```typescript
// e2e/arc/tests/smoke.spec.ts
import { test, expect } from '@playwright/test';

test('arc home page loads', async ({ page }) => {
  await page.goto('http://localhost:4201');
  await expect(page.getByRole('heading')).toHaveText(/Arc Home/i);
});
```

Similar is needed for `saas-ui` to avoid silent failures.

**2. Page Object Pattern**

Shared selectors and repeated logic should move into page objects. Example structure:

```typescript
// e2e/arc/page-objects/header.po.ts
export class Header {
  constructor(private readonly page) {}
  logo() { return this.page.getByRole('img', { name: 'Logo' }); }
  navLink(name: string) { return this.page.getByRole('link', { name }); }
}
// Usage:
import { Header } from '../page-objects/header.po';
const header = new Header(page);
await expect(header.logo()).toBeVisible();
```

**3. Negative/Edge Case Coverage**

All tests currently cover only positive paths. Please include cases for 404s, invalid inputs, route guards, and error states. Example:

```typescript
test('shows 404 for unknown page', async ({ page }) => {
  await page.goto('http://localhost:4201/non-existent-page');
  await expect(page.getByText('Page Not Found')).toBeVisible();
});
```

**4. CI/CD Integration**

No CI config is provided for E2E. Recommend adding GitHub Actions that runs Playwright, collects traces/videos, and fails the build on test errors.

**5. Test Naming**

Add and enforce clear naming conventions for test/spec files, e.g. `feature.spec.ts` for easier tracking in reports.

**Potential Issues**

- Parallel dev server runs may cause port collisions if ports are reused.
- Retry logic is basic. If some tests are flaky, consider backoff, more explicit waits, or separating longer-running flows.

**Summary:**

Expand coverage for all projects, build out shared page objects, include negative testing, and clarify CI handling. These changes will enhance reliability and help the E2E suite scale with the monorepo.

File: package.json [hunk 1]
```diff
@@ -4,11 +4,18 @@
...
     "build": "ng build",
     "test": "ng test",
     "lint": "eslint projects/**/*.ts",
     "lint:fix": "npm run lint -- --fix",
-    "e2e": "ng e2e",
+    "e2e": "playwright test --config=e2e/arc-docs/playwright.config.ts",
+    "e2e:all": "npm run e2e:arc-docs && npm run e2e:arc && npm run e2e:saas-ui",
+    "e2e:arc-docs": "playwright test --config=e2e/arc-docs/playwright.config.ts",
+    "e2e:arc": "playwright test --config=e2e/arc/playwright.config.ts",
+    "e2e:headed": "playwright test --headed --config=e2e/arc-docs/playwright.config.ts",
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply package.json]: removed

File: e2e/arc-docs/tests/arc-docs-test-plan.md [hunk 2]
```diff
@@ -0,0 +1,782 @@
...
+
+1. Navigate to `http://localhost:4200/`
+2. Expand "Getting Started" and click "Introduction"
+3. Expand "Guide" and click "Cloning Boilerplate"
+4. Click "Backend Integration"
+5. Expand "Auth" and navigate through all Auth pages in order:
+   - Introduction
+   - Installation
+   - Getting User Token
+   - Setting Up Api Configration
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply e2e/arc-docs/tests/arc-docs-test-plan.md]: fixed

File: playwright.config.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ *
+ * To run all tests: npm run e2e:all
+ *
+ * See https://playwright.dev/docs/test-configuration.
+ */
+export default defineConfig({
+  // This config is intentionally minimal as project-specific configs are in subdirectories
+  // You can run tests from specific project directories or use the npm scripts
+
+  testDir: './e2e',
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply playwright.config.ts:24]: kept for reference

File: package.json [hunk 4]
```diff
@@ -4,11 +4,18 @@
...
+    "start:saas-ui": "ng serve saas-ui --port 4202",
     "build": "ng build",
     "test": "ng test",
     "lint": "eslint projects/**/*.ts",
     "lint:fix": "npm run lint -- --fix",
-    "e2e": "ng e2e",
+    "e2e": "playwright test --config=e2e/arc-docs/playwright.config.ts",
+    "e2e:all": "npm run e2e:arc-docs && npm run e2e:arc && npm run e2e:saas-ui",
+    "e2e:arc-docs": "playwright test --config=e2e/arc-docs/playwright.config.ts",
+    "e2e:arc": "playwright test --config=e2e/arc/playwright.config.ts",
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply package.json:16]: added

File: e2e/arc-docs/tests/arc-docs-test-plan.md [hunk 5]
```diff
@@ -0,0 +1,782 @@
...
+1. Navigate to `http://localhost:4200/`
+2. Expand "Auth" menu
+3. Click on "Setting Up Api Configration"
+4. Verify page loads
+
+**Expected Results:**
+
+- URL changes to `/docs/auth-doc/configureEnvs`
+- Page content loads successfully
+- "Setting Up Api Configration" menu item is highlighted as active
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply e2e/arc-docs/tests/arc-docs-test-plan.md]: fixed

[the remark this answers is not in the document]
Comment (DeepakSF25) [reply e2e/arc-docs/tests/arc-docs-test-plan.md]: fixed

File: .vscode/mcp.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "servers": {
+    "playwright-test": {
+      "type": "stdio",
+      "command": "npx",
+      "args": [
+        "playwright",
+        "run-test-mcp-server"
```
[the remark this answers is not in the document]
Comment (DeepakSF25) [reply .vscode/mcp.json:8]: false positive

**Review by yeshamavani — APPROVED** [review]

**Review by rohit-sourcefuse — APPROVED** [review]
